discretisedfield/tests/test_field.py

```diff
@@ -3085,6 +3085,26 @@ def test_mpl_vector(test_field):
     plt.close("all")
 
 
+@pytest.mark.parametrize("nvdim", [1, 2, 3, 4])
+def test_mpl_dimension_vector(valid_mesh, nvdim):
+    field = df.Field(valid_mesh, nvdim=nvdim)
+
+    if valid_mesh.region.ndim != 2:
+        with pytest.raises(RuntimeError):
+            field.mpl.vector()
+    else:
+        if nvdim != 2:
+            with pytest.raises(ValueError):
+                field.mpl.vector()
```

Comment:
`mpl.vector` also works for `nvdim=3`. We use colour to show the out-of-plane component. Additionally, any higher `nvdim` is also allowed if `vdim_mapping` is defined (the latter is also required for `nvdim in [1, 2, 3]`). For high-dimensional fields we do not do automatic colouring and only show the two components in the plot plane obtained from `vdim_mapping` (or specified explicitly using the `vdims` argument of `mpl.vector`).